api-server/src/server/middlewares/csurf-error-handler.js

```diff
@@ -0,0 +1,12 @@
+import { csrfOptions } from './csurf.js';
+
+export default function csrfErrorHandler() {
+  return function (err, req, res, next) {
+    if (err.code === 'EBADCSRFTOKEN') {
+      // use the middleware to generate a token. The client sends this back via
+      // a header
+      res.cookie('csrf_token', req.csrfToken(), csrfOptions);
```

Comment:
I am not sure why are setting a cookie when the CSRF verification fails.

Reply:
If this isn't done, then a user can never recover from having a bad cookie/token.  Without it the server does this

1) Does this header token come from the secret?
2) If not, deny request.

That's it.  Because this cookie setting logic is in a separate middleware, it never gets called and the user never gets in.

By adding this error handling, if the user supplies an invalid token, then they will be issued a valid one.  Their next attempt should succeed.